pageserver/src/tenant/storage_layer/layer.rs

```diff
@@ -1207,23 +1210,31 @@ impl DownloadedLayer {
                 ));
                 image_layer::ImageLayerInner::load(&owner.path, lsn, summary, ctx)
                     .await
-                    .map(LayerKind::Image)
-            }
-            // this will be a permanent failure
-            .context("load layer");
+                    .map(|res| res.map(LayerKind::Image))
+            };
 
-            if let Err(e) = res.as_ref() {
-                LAYER_IMPL_METRICS.inc_permanent_loading_failures();
-                // TODO(#5815): we are not logging all errors, so temporarily log them here as well
-                tracing::error!("layer loading failed permanently: {e:#}");
+            match res {
+                Ok(Ok(layer)) => Ok(Ok(layer)),
+                Ok(Err(transient)) => Err(transient),
+                Err(permanent) => {
+                    LAYER_IMPL_METRICS.inc_permanent_loading_failures();
+                    // TODO(#5815): we are not logging all errors, so temporarily log them **once**
+                    // here as well
+                    tracing::error!("layer loading failed permanently: {permanent:#}");
+                    Ok(Err(permanent))
+                }
             }
```

Comment:
```suggestion
            res.map_err(|permanent| {
            LAYER_IMPL_METRICS.inc_permanent_loading_failures();
            // TODO(#5815): we are not logging all errors, so temporarily log them **once**
            // here as well
            tracing::error!("layer loading failed permanently: {permanent:#}");
            permanent
        })
```

Reply:
The point of this PR is to separate transient and permanent errors, this would essentially be the `main` version before this PR where all errors are permanent.